steps/discover-repos.rb

```diff
@@ -97,44 +82,110 @@ def mock_reps(page, size, licenses)
   }
 end
 
-def cooldown(opts, found)
-  puts "Let's sleep for #{opts[:pause]} seconds to cool off GitHub API \
-(already found #{found.count} repos, need #{opts[:total]})..."
-  sleep opts[:pause]
+def process_year(year, github, context)
+  query = build_query(year, context[:opts])
+  puts "Querying for repositories created in #{year}..."
+
+  loop_through_pages(query, github, context)
+  puts "Completed querying for year #{year}. Found #{context[:found].count} repositories so far."
 end
 
-puts 'Not searching GitHub API, using mock repos' if opts[:dry]
-loop do
-  break if page * size > max
-  count = 0
-  json = if opts[:dry]
-    mock_reps(page, size, licenses)
+def build_query(year, opts)
+  [
+    "stars:#{opts['min-stars']}..#{opts['max-stars']}",
+    "size:>=#{opts['min-size']}",
+    'language:java',
+    "created:#{year}-01-01..#{year}-12-31",
+    'is:public',
+    'mirror:false',
+    'archived:false', 'template:false', 'NOT', 'android'
+  ].join(' ')
+end
+
+def loop_through_pages(query, github, context)
+  page = 0
+  loop do
+    break if context[:found].count >= context[:opts][:total]
+
```

Comment:
@lueFlake better avoid them (empty lines)

Reply:
@yegor256 removed blank lines, forgot to remove them after debug, my bad